Library/Homebrew/formula_installer.rb

```diff
@@ -608,6 +610,11 @@ def finish
     ohai "Summary" if verbose? || show_summary_heading?
     puts summary
 
+    # Updates the cache for a particular formula after doing an install
+    DatabaseCache.new(:linkage) do |database_cache|
+      LinkageChecker.new(keg, database_cache, false, formula) unless database_cache.empty?
```

Comment:
@AndrewMcBurney Presumably this doesn't create the file on disk if `database_cache.empty?`, right? This line is a bit on the long side now; may want to use the `unless ... end` form or `break if database_cache.empty?` on the line above.

Reply:
No, it will create an empty database cache if the file does not exist on [this line](https://github.com/AndrewMcBurney/brew/blob/cache-optimization/Library/Homebrew/cache_store.rb#L24) when `DatabaseCache.new(:linkage)` is called.

Reply:
@AndrewMcBurney Gotcha. I think it would be good if e.g.

```ruby
   DatabaseCache.new(:linkage) do |database_cache|
      use_cache = !ENV["HOMEBREW_LINKAGE_CACHE"].nil?
      checker = LinkageChecker.new(keg, database_cache, use_cache, formula)
```

could be refactored so the cache isn't created unless `ENV["HOMEBREW_LINKAGE_CACHE"]` (or the equivalent command-line argument) is specified. It seems strange to create a cache file on disk but not populate or access it otherwise. An option for this may be lazy cache creation where the file is only created the first time you attempt to write to it.